### PR TITLE
chore: strict ts errors

### PR DESCRIPTION
**Pull request by PatrykBuniX**
Fixing ts strict errors

### CODE REVIEW

File: src/script/components/panel/PanelActions/PanelActions.tsx [hunk 1]
```diff
@@ -37,18 +37,19 @@ export interface PanelActionsProps {
...
-              <Icon name={icon} />
-            </span>
-            <span data-uie-name={`${identifier}-item-text`} className="panel__action-item__text">
-              {label}
-            </span>
-          </button>
-        </li>
-      ))}
+      {items &&
+        items.map(({click, identifier, icon, label}) => (
```
Comment (atomrc) [inline src/script/components/panel/PanelActions/PanelActions.tsx]: Maybe we could make the items not optional. Does it really make sense to have them this way?!

File: src/script/components/userDevices/DeviceCard.test.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ describe('DeviceCard', () => {
       device: createClientEntity({
         class: ClientClassification.DESKTOP,
         meta: {
-          isVerified: ko.observable(false),
+          isVerified: ko.observable(false as boolean),
```
Comment (atomrc) [inline src/script/components/userDevices/DeviceCard.test.ts]: humm what is the error here, this sounds weird

File: src/script/page/message-list/MentionSuggestions/MentionSuggestionsItem.tsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ const MentionSuggestionsItem: React.ForwardRefRenderFunction<HTMLDivElement, Men
     ['name', 'expirationRemainingText', 'isTemporaryGuest', 'isExternal', 'isDirectGuest'],
   );
 
-  const onClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+  const onClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (atomrc) [inline src/script/page/message-list/MentionSuggestions/MentionSuggestionsItem.tsx]:
```suggestion
  const onClick = (event: React.UIEvent) => {
```
As we never use any specificity of `Keyboard` or `Mouse` event, we can go for a more generic type

File: src/script/components/MessagesList/Message/ContentMessage/asset/AssetLoader.tsx [hunk 4]
```diff
@@ -33,7 +33,7 @@ const AssetLoader: React.FC<AssetLoaderProps> = ({large, loadProgress, onCancel}
   const progress = `${loadProgress * elementScale} ${100 * elementScale}`;
   const viewBoxSize = 32 * elementScale;
   const viewBox = `0 0 ${viewBoxSize} ${viewBoxSize}`;
-  const onClick = (event: React.MouseEvent<HTMLDivElement>) => {
+  const onClick = (event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (atomrc) [inline src/script/components/MessagesList/Message/ContentMessage/asset/AssetLoader.tsx]:
```suggestion
  const onClick = (event: React.UIEvent) => {
```

File: src/script/components/CopyToClipboard.tsx [hunk 5]
```diff
@@ -26,12 +26,18 @@ export interface CopyToClipboardProps {
 }
 
 const CopyToClipboard: React.FC<CopyToClipboardProps> = ({text}) => {
-  const onClick = ({currentTarget}: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+  const onClick = ({
+    currentTarget,
+  }: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (atomrc) [inline src/script/components/CopyToClipboard.tsx]:
```suggestion
  const onClick = ({currentTarget}: React.UIEvent) => {
```

File: src/script/components/AvailabilityState.tsx [hunk 6]
```diff
@@ -73,10 +73,12 @@ const AvailabilityState: React.FC<AvailabilityStateProps> = ({
...
-        ...event,
-        nativeEvent: newEvent,
-      } as unknown as React.MouseEvent<Element, MouseEvent>);
+      if (onClick) {
+        onClick({
+          ...event,
+          nativeEvent: newEvent,
+        } as unknown as React.MouseEvent<Element, MouseEvent>);
+      }
     }
```
Comment (atomrc) [inline src/script/components/AvailabilityState.tsx]:
```suggestion
        onClick?.({
          ...event,
          nativeEvent: newEvent,
        } as unknown as React.MouseEvent<Element, MouseEvent>);
      }
```

File: src/script/components/Modals/ServiceModal/ServiceModal.tsx [hunk 7]
```diff
@@ -43,7 +43,9 @@ const ServiceModal: React.FC<ServiceModalProps> = ({
   onClose,
 }) => {
   const onOpenService = () => {
-    onClose();
+    if (onClose) {
+      onClose();
+    }
```
Comment (atomrc) [inline src/script/components/Modals/ServiceModal/ServiceModal.tsx]:
```suggestion
      onClose?.();
```

File: src/script/components/calling/DeviceToggleButton.tsx [hunk 8]
```diff
@@ -28,7 +28,9 @@ export interface DeviceToggleButtonProps {
 }
 
 const DeviceToggleButton: React.FC<DeviceToggleButtonProps> = ({currentDevice, devices, onChooseDevice, styles}) => {
-  const selectNextDevice = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+  const selectNextDevice = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>,
```
Comment (atomrc) [inline src/script/components/calling/DeviceToggleButton.tsx]:
```suggestion
    event: React.UIEvent,
```

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/script/components/panel/PanelActions/PanelActions.tsx (hunk 1 above)
Comment (PatrykBuniX) [reply src/script/components/panel/PanelActions/PanelActions.tsx]: I agree and eventually render the component conditionally. There is no need to render it when there are no items.

File: src/script/components/userDevices/DeviceCard.test.ts (hunk 2 above)
Comment (PatrykBuniX) [reply src/script/components/userDevices/DeviceCard.test.ts]: Very strange error, that 'false' (as if it was casted like: false as const) is not accessible to boolean. (`Type 'Observable<false>' is not assignable to type 'Observable<boolean>'.`)

File: src/script/page/message-list/MentionSuggestions/MentionSuggestionsItem.tsx (hunk 3 above)
Comment (PatrykBuniX) [reply src/script/page/message-list/MentionSuggestions/MentionSuggestionsItem.tsx]: Sure, didn't know that such a type exists, I was always explicitly typing it. Makes a lot of sense.

Comment (PatrykBuniX) [reply src/script/page/message-list/MentionSuggestions/MentionSuggestionsItem.tsx]: I've checked it and we're always passing an array to props (sometimes empty), so definitely no reason to make it optional here ;)

File: src/script/components/userDevices/DeviceCard.test.ts (hunk 2 above)
Comment (PatrykBuniX) [reply src/script/components/userDevices/DeviceCard.test.ts]: Changed to `ko.observable<boolean>(false)`

Comment (PatrykBuniX) [reply src/script/components/userDevices/DeviceCard.test.ts]: Generic was inferred from passed value, thats why this error appeared 😄

**Review by atomrc — APPROVED** [review]

**Pull request merged**